SConstruct

```diff
@@ -1578,6 +1579,24 @@ if env['python_package'] != 'none':
             print('INFO: Building the full Python package for Python {0}'.format(python_version))
             env['python_package'] = 'full'
 
+if env["python_package"] == "full" and env["OS"] == "Darwin":
+    # sysconfig.get_platform() looks something like: macosx-11.0-arm64
+    # MACOSX_DEPLOYMENT_TARGET should be the same as or lower than
+    # the one set when Python was compiled to ensure ABI compatibility.
+    # If it's been set in the user's environment and passed through env_vars
+    # configuration, use that value instead.
+    if not env["ENV"].get("MACOSX_DEPLOYMENT_TARGET", False):
+        info = get_command_output(
+            env["python_cmd"],
+            "-c",
+            "import sysconfig; print(sysconfig.get_platform())"
+        )
+        env["ENV"]["MACOSX_DEPLOYMENT_TARGET"] = info.split("-")[1]
+
+    env.Append(
+        CXXFLAGS=f"-mmacosx-version-min={env['ENV']['MACOSX_DEPLOYMENT_TARGET']}"
+    )
+
```

Comment:
This is a decently significant change of behavior for compiling on macOS. As shown here, I scoped it to only be done when the Python package is being built, since that's the main case where we want to be more careful about ABI compatibility I think. But if you think this should be a more general change, that's fine too.

Reply:
This is likewise clear-as-mud to me. Since I don't think any of us understands this (and maybe no one outside compiler experts at Apple do), I'm happy to drop this change. I'm happy to admit that I was trying to copy some of the conda-forge magic from their recipes in here, but I don't understand it, so I'll just drop it.